### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #11 [May] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#9524] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, May 25th (9am PDT/6pm CEST)**
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/9426
- Recording:

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [x] Encourage additions to the agenda
- [x] Publicise on forum and Discord
- [ ] Appoint secretary (to take notes in Github comment section ✍️)
- [ ] RECORD meeting 🎥

**Comment by arthurgousset** [comment]
@aaronmboyd kindly agreed to facilitate this meeting. I'm OOO around those days :)

**Reply by aaronmboyd** [reply]
I did not ☺️ But perhaps another Aaron did @aaronmgdr ?

**Reply by arthurgousset** [reply]
My bad, sorry for the ping @aaronmboyd! It's @aaronmgdr indeed 🙌

**Comment by aaronmgdr** [comment]
Launch of React-Celo Beta (Successor to use-contractkit)

**Reply by andyle83** [reply]
Does this will limit options if developers prefer other UI framework / lib ?

**Reply by aaronmgdr** [reply]
not anymore than status quo as this is a renaming of `use-contractkit` which was already react specific. (now its just more clear)

**Reply by aaronmgdr** [reply]
Theming API 
React 17 and 18
Bug Fixes
compiled to es6 (not es5)
Built more automated tests

**Comment by aaronmgdr** [comment]
New Moola Dashboard Screen

**Reply by aaronmgdr** [reply]
https://makerdojo.io/public/dashboards/tWnDs1CsMQyaSLHFHj0MUoNz20WbxjGv72fCTxgq?org_slug=default

**Comment by aaronmgdr** [comment]
A tool To debug wallet connect issues between wallets and dapps: (@jeanregisser )

**Reply by aaronmgdr** [reply]
dan: we started something on this

eric: like a tool to monitor errors? 

Jean: More like a tool to connect to and test operations

eric: for users or for developers? 

jean: first for users to give better info to developers, could also be used for devs to test implementation. havent seen list of things in wc docs would could be supported. could it be source of truth?  maybe one for wallets and one for dapps?  


Primus Money: may not be at the stage, when using wc we dont know where the issue issue. need debugger tools. to know where error is happening. Is it on server? wallet? dapp?  pinpoint the trouble spot. can we prove fault? 


Jean:  good point about WC sockets. no status page I know of.  could we talk with them? 

Primus Money: Could there be a log? Debug Mode in react-celo? 

DAN: Sharing idea of testing wallet test plan. Connect your wallet and see if what works and what issues come up

Jean: Looks Great.  | Logs would be great yes. Valora has logs. maybe we enable dev mode? 

Primus: Requirements: initiate connection, better infor on where time is lost? show more user or dev info about what is happening. 

Dan: I created an issue for this, might get int v4.1

**Reply by aaronmgdr** [reply]
Primus Money: Recovering from previous connect is a mess. Better to start over

**Reply by heymateag** [reply]
Anil Kutty: We have had struggles with integrating our DApp with WC (with our own non-custodial wallet as well as Valora) and are having to implement a web 2.5 alternative version of using mobile push to trigger purchase internally within our own mobile wallet. We hosted a relay server ourselves instead of relying walletconnect.com, extra logging on our mobile wallet clients but still no luck. Sometimes it appears that the WC socket calls might be blocked by some ISPs, it's a hypothesis for which we don't have an evidence yet. 

Just an idea - Host a WalletConnect relay server for the Celo community (maybe the Valora team can do that as they see most WC traffic in the community) and change the logger in the  WC implementation to get more detailed logging to be able to trace transactions end to end with timestamps. We will need some place for everyone to be able to view these logs and search for their transactions. If no one has time to change the WC code, ask the WC team for enhanced logging feature in their near-term roadmap.
